File: training-data/zig/floyd_warshall.zig
const std = @import("std");

const INF: i32 = 999999;

fn floydWarshall(dist: [][4]i32) void {
    const n = dist.len;
    for (0..n) |k| {
        for (0..n) |i| {
            for (0..n) |j| {
                if (dist[i][k] != INF and dist[k][j] != INF and dist[i][k] + dist[k][j] < dist[i][j]) {
                    dist[i][j] = dist[i][k] + dist[k][j];
                }
            }
        }
    }
}

pub fn main() void {
    var dist = [4][4]i32{
        .{ 0, 3, INF, 7 },
        .{ 8, 0, 2, INF },
        .{ 5, INF, 0, 1 },
        .{ 2, INF, INF, 0 },
    };
    floydWarshall(&dist);
    for (dist) |row| {
        std.debug.print("{any}\n", .{row});
    }
}

File: training-data/swift/floyd_warshall.swift
func floydWarshall(_ n: Int, _ edges: [(Int, Int, Int)]) -> [[Int]] {
    let inf = Int.max / 2
    var dist = [[Int]](repeating: [Int](repeating: inf, count: n), count: n)
    for i in 0..<n { dist[i][i] = 0 }
    for (u, v, w) in edges { dist[u][v] = w }

    for k in 0..<n {
        for i in 0..<n {
            for j in 0..<n where dist[i][k] + dist[k][j] < dist[i][j] {
                dist[i][j] = dist[i][k] + dist[k][j]
            }
        }
    }
    return dist
}

let edges = [(0, 1, 3), (0, 2, 8), (1, 2, 2), (2, 3, 1), (3, 0, 4)]
for row in floydWarshall(4, edges) {
    print(row)
}
